pkg/router/template/plugin.go

```diff
@@ -360,7 +362,13 @@ func createRouterEndpoints(endpoints *kapi.Endpoints, excludeUDP bool, lookupSvc
 				s := ep.ID
 				ep.IdHash = fmt.Sprintf("%x", md5.Sum([]byte(s)))
 
-				out = append(out, ep)
+				// Add only not duplicated endpoints.
+				if !duplicated[ep.ID] {
+					out = append(out, ep)
+					duplicated[ep.ID] = true
+				} else {
+					log.V(4).Info("skip a duplicated endpoints to add", ep.ID)
```

Comment:
nit: the message doesn't make sense.  Maybe "skipped adding a duplicated endpoints object"
/lgtm

Reply:
In https://github.com/openshift/router/pull/285, @frobware and I noticed this but decided it wasn't worth lamenting over.